components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -252,16 +310,22 @@ public boolean keepAlive() {
     }
 
     /**
-     * Checks if the request has been transferred over a secure connection. If the protocol is HTTPS and the
-     * content is delivered over SSL then the request is considered to be secure.
+     * Checks if the request has been transferred over a secure connection.
+     * If the protocol is HTTPS then the request is considered to be secure.
      *
-     * @return true if the request is transferred securely
+     * @return true if protocol is set to "https".
      */
     public boolean isSecure() {
         return secure;
     }
 
+    /**
```

Comment:
There should be an `@deprecated` tag in the Javadoc. "The @deprecated description in the first sentence should at least tell the user when the API was deprecated and what to use as a replacement." [Source](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#@deprecated)